news-explorer-frontend/src/components/App/App.jsx

```diff
@@ -0,0 +1,342 @@
+  const [isCheckingToken, setIsCheckingToken] = useState(true)
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
+  const [isSignUpPopupOpen, setIsSignUpPopupOpen] = useState(false);  // isLoggedIn & isSignUpPopupOpen
+  const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
+  const [isMobileClicked, setIsMobileClicked] = useState(false);
+  const [isMobileMenuClicked, setIsMobileMenuClicked] = useState(false); 
+  const [filteredResults, setFilteredResults] = useState([]);
+  const [keyword, setKeyword] = useState('');
+
+    //MainApi signup
+    const onRegisterUser = ({ name, email, password }) => {
+      api.signup({name, email, password})
+        // .then((res) => {
+        //   if(res._id) {
+        //     navigate('/signin')
+        //   }
+        // })
+        // .then(() => {
+        //   setIsSignInPopupOpen(false)
+        //   setIsRegisterPopupOpen(true)})
+        .catch((err) => {
+          console.log("signup err =>", `Error: ${err.status}`);
+        })
+        .finally(() => {
+          closeAllPopups();
+          setIsRegisterPopupOpen(true);
+        })
+    }
+
+    // MainApi Signin
+    const onLogin = ({  email, password }) => {
+      api.signin({email, password})
+        .then((res) => {
+          if(res.token) {
+            setIsLoggedIn(true);
+            setUserData(res.name); // check if I need to use this later on
+            localStorage.setItem('jwt', res.token);
+            navigate('/');
+          } else {
+            console.log("signin fail")
+          }
+        })
+        .catch((err) => {
+          console.log("signin fail=>", err);
+        })
+        .finally(() => closeAllPopups())
+    }
+
+    //signout
+    function handlesignOut() {
+      setIsLoggedIn(false);
+      localStorage.removeItem('jwt');
+      navigate('/');
+    }
+
+    //MainApi checkToken
+    useEffect(() => {
+      const token = localStorage.getItem('jwt')
+      if(token) {
+        api.checkToken(token)
+        .then((res) => {
+          if(res.data._id) {
+            setIsLoggedIn(true);
+            setUserData({ email: res.data.email})
+            navigate('/')
+          } else {
+            navigate('/signin');
+            localStorage.removeItem(token);
+          }
+        })
+        .catch((err) => {
+          console.log("err =>", err)
+          navigate('/signin')
+        })
+        .finally(() => setIsCheckingToken(false))
+      }
+    }, [])
+
+    
+    useEffect(() => {
+      const token = localStorage.getItem('jwt')
+      //console.log("user", currentUser._id)
+      if( token && currentUser._id) {
+        api.getArticles()
+          .then(res => {
+            const savedArticlesFiltered = res.data.filter(article => article.owner == currentUser._id)
+            setSavedArticle(savedArticlesFiltered)
+          })

```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**

Reply:
checked and added catch on every missing  request